### PR TITLE
Fix jvm crash during tests

### PR DESCRIPTION
**Pull request by ilfrin**
those two were seen to cause intermittent jvm crashes like below, during TPCDS product tests

```
2017-12-02 16:58:00 INFO: [1 of 10] sql_tests.testcases.tpcds.q80 (Groups: tpcds)
presto-master_1       | #
presto-master_1       | # A fatal error has been detected by the Java Runtime Environment:
presto-master_1       | #
presto-master_1       | #  SIGSEGV (0xb) at pc=0x00007f39eda79241, pid=6, tid=0x00007f39dd2f8700
presto-master_1       | #
presto-master_1       | # JRE version: Java(TM) SE Runtime Environment (8.0_152-b16) (build 1.8.0_152-b16)
presto-master_1       | # Java VM: Java HotSpot(TM) 64-Bit Server VM (25.152-b16 mixed mode linux-amd64 compressed oops)
presto-master_1       | # Problematic frame:
presto-master_1       | # V  [libjvm.so+0x692241]  KlassToOopClosure::do_klass(Klass*)+0x11
presto-master_1       | #
presto-master_1       | # Core dump written. Default location: /var/presto/core or core.6
presto-master_1       | #
presto-master_1       | # An error report file with more information is saved as:
presto-master_1       | # /docker/volumes/logs/product-tests-presto-jvm-error-file.log
presto-master_1       | #
```

### CODE REVIEW

File: presto-product-tests/conf/presto/etc/jvm.config [hunk 1]
```diff
@@ -10,13 +10,10 @@
...
--XX:+CMSClassUnloadingEnabled
--XX:+AggressiveOpts
 -XX:+HeapDumpOnOutOfMemoryError
 -XX:+ExitOnOutOfMemoryError
 -XX:ReservedCodeCacheSize=150M
 -DHADOOP_USER_NAME=hive
 -Duser.timezone=Asia/Kathmandu
--Xdebug 
--Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=5005
+#-Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=5005
```
Comment (findepi) [inline presto-product-tests/conf/presto/etc/jvm.config]: Just remove?

File: presto-product-tests/conf/presto/etc/jvm.config [hunk 2]
```diff
@@ -10,8 +10,6 @@
 -XX:+UseG1GC
 -XX:+UseGCOverheadLimit
 -XX:+ExplicitGCInvokesConcurrent
--XX:+CMSClassUnloadingEnabled
```
Comment (findepi) [inline presto-product-tests/conf/presto/etc/jvm.config]: Commit msg doesn't conform (sentence case). Btw, what about "Do not use unrecommended JVM opts in product tests" ?

File: presto-product-tests/conf/presto/etc/jvm.config [hunk 3]
```diff
@@ -15,6 +15,5 @@
 -XX:ReservedCodeCacheSize=150M
 -DHADOOP_USER_NAME=hive
 -Duser.timezone=Asia/Kathmandu
--Xdebug
```
Comment (findepi) [inline presto-product-tests/conf/presto/etc/jvm.config]: Commit message
* sentence case
* what about "Do not use JVM debug options when running product tests"

also, you may want to remove `-Xdebug` in separate commit labelling eg "Remove no-op JVM flag from product tests execution"

File: presto-product-tests/conf/presto/etc/jvm.config (hunk 2 above)
Comment (electrum) [reply presto-product-tests/conf/presto/etc/jvm.config]: Does this option do anything with G1? Assuming it does not, then removing `AggressiveOpts` is the only real change here, so we should do that separately with a commit message to reflect it. Something like 
*"Remove AggressiveOpts JVM flag for product tests"*

Comment (electrum) [reply presto-product-tests/conf/presto/etc/jvm.config]: I just checked and we don't use `AggressiveOpts` on our production clusters, either

File: presto-product-tests/conf/presto/etc/jvm.config (hunk 3 above)
Comment (electrum) [reply presto-product-tests/conf/presto/etc/jvm.config]: How about *"Do not enable JVM debugger when running product tests"*

File: presto-product-tests/conf/presto/etc/jvm.config (hunk 2 above)
Comment (sopel39) [reply presto-product-tests/conf/presto/etc/jvm.config]: It was the agent that caused failure.

File: presto-product-tests/conf/presto/etc/jvm.config (hunk 3 above)
Comment (findepi) [reply presto-product-tests/conf/presto/etc/jvm.config]: @electrum note that https://docs.oracle.com/javase/8/docs/technotes/tools/unix/java.html lists `-Xdebug` as "Does nothing. Provided for backward compatibility.", so removal of `-Xrunjdwp:transport:..` is the most important change here

File: presto-product-tests/conf/presto/etc/jvm.config (hunk 1 above)
Comment (ilfrin) [reply presto-product-tests/conf/presto/etc/jvm.config]: I wanted to leave that in for ease of use when someone locally wants to debug it. I don't care much though.

Comment (findepi) [reply presto-product-tests/conf/presto/etc/jvm.config]: Nice. Then I'd recommend updating to modern form (`-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005`). But i prefer removal anyway :)

**Comment by ilfrin** [comment]
AC

File: presto-product-tests/conf/presto/etc/jvm.config (hunk 3 above)
Comment (ilfrin) [reply presto-product-tests/conf/presto/etc/jvm.config]: done

File: presto-product-tests/conf/presto/etc/jvm.config (hunk 2 above)
Comment (ilfrin) [reply presto-product-tests/conf/presto/etc/jvm.config]: done

**Comment by nezihyigitbasi** [comment]
@ilfrin Which commit actually fixes the problem here? Is it only the agent change as @sopel39 comments above or do we need other changes? Can you please update the PR to keep only the relevant change(s) and squash your commits?

**Comment by findepi** [comment]
@nezihyigitbasi , it's the removal of `-Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=5005` which fixes the product tests JVM crashes, but we should merge all changes, as they clean up other things, like removing no-op flags or flags we should not turn on in tests.
Btw, initially, they were squashed, but they are separate now now to give false impression that e.g. removal of `AggresiveOpts` was (part of) the fix.

**Comment by nezihyigitbasi** [comment]
@findepi thanks for the explanation.

**Pull request merged**